File: tests/fixtures/positive/security/OkHttpDisableSslValidation.java
package test;

import okhttp3.OkHttpClient;
import javax.net.ssl.SSLSocketFactory;
import javax.net.ssl.X509TrustManager;

class ClientFactory {
    OkHttpClient verifier() {
        return new OkHttpClient.Builder()
            .hostnameVerifier((hostname, session) -> true)
            .build();
    }

    OkHttpClient trustManager(SSLSocketFactory socketFactory, X509TrustManager unsafeTrustManager) {
        return new OkHttpClient.Builder()
            .sslSocketFactory(socketFactory, unsafeTrustManager)
            .build();
    }
}

File: tests/fixtures/positive/security/OkHttpDisableSslValidation.kt
package test

import okhttp3.OkHttpClient
import javax.net.ssl.SSLSocketFactory
import javax.net.ssl.X509TrustManager

class ClientFactory {
    fun verifier(): OkHttpClient {
        return OkHttpClient.Builder()
            .hostnameVerifier { _, _ -> true }
            .build()
    }

    fun trustManager(socketFactory: SSLSocketFactory, unsafeTrustManager: X509TrustManager): OkHttpClient {
        return OkHttpClient.Builder()
            .sslSocketFactory(socketFactory, unsafeTrustManager)
            .build()
    }
}
